Assets/LeapMotion/Core/Scripts/Attributes/ImplementsInterface.cs

```diff
@@ -55,35 +57,78 @@ public class ImplementsInterfaceAttribute : CombinablePropertyAttribute,
     /// the interface that this attribute constrains objects to, and returns the object
     /// that implements that interface, or null if none was found.
     /// </summary>
-    public UnityObject FindImplementer(UnityObject obj) {
-
-      if (!fieldInfo.FieldType.IsAssignableFrom(obj.GetType())
-          && !(typeof(Component).IsAssignableFrom(fieldInfo.FieldType)
-               && obj.GetType() == typeof(GameObject))) {
-        // Even if the object implements the correct interface, the field isn't
-        // compatible with this object. E.g. A ScriptableObject can't be assigned to a
-        // MonoBehaviour field.
+    public UnityObject FindImplementer(SerializedProperty property, UnityObject fromObj) {
+
+      // Don't use fieldInfo here because it is very convenient for it to be left null
+      // on the CombinablePropertyAttribute when dealing with generic-type situations
+      // where the ImplementsInterface class has to be constructed manually in a custom
+      // editor. (Specific case: StreamConnectorEditor.cs)
+
+      bool isTypeDefinitelyIncompatible;
+      {
+        isTypeDefinitelyIncompatible = !this.type.IsAssignableFrom(fromObj.GetType());
+
         // We have to make an exception when a GameObject is dragged into a field whose
-        // type is a Component; we use GetComponent to satisfy that case.
+        // type is a Component, because it's expected that we can use GetComponent in
+        // that case. (We might still fail later if the component isn't found.)
+        var objIsGameObject = fromObj.GetType() == typeof(GameObject);
+        if (objIsGameObject) {
+          isTypeDefinitelyIncompatible = false;
+        }
+
+        // We also make an exception when a Component is dragged into a field that
+        // doesn't directly satisfy the interface because it might have other Components
+        // on its GameObject that _do_ satisfy the field, again via GetComponent.
+        var objIsComponent = typeof(Component).IsAssignableFrom(fromObj.GetType());
+        if (objIsComponent) {
+          isTypeDefinitelyIncompatible = false;
+        }
+
+        // However, you can't assign a ScriptableObject to a field expecting a Component,
+        // or vice-versa, no matter what.
+
+        // Matches the "Foo" in "PPtr<$Foo>" (as an inner group)
+        var fieldTypeMatch = Regex.Match(property.type, "PPtr<\\$(.+)>");
+        if (fieldTypeMatch.Groups.Count == 0) {
```

Comment:
The if clause here assigns 'true' to the variable `isTypeDefinitelyIncompatible`.  In the else clause, the variable is assigned to if the field is either takes a scriptable object, or of it takes a component.  But those are the only two possibilities, and so `isTypeDefinitelyIncompatible` will always be assigned to, making the 3 assignments above (line 69, 76, and 84) not necessary.